andela/travel_tool_back
# #160370645 Create and edit requests with status 'open'

#### What does this PR do?
Allows a user to create any type of request and also edit an open request

#### Description of Task to be completed?
- Modify the endpoint for creating a request so that users can make oneWay, multi and return trips.
- Create an endpoint for editing open requests
- Write tests for the endpoint
- Create validations for the endpoints

#### How should this be manually tested?
- Clone the repo `git clone https://github.com/andela/travel_tool_back.git`
- Run `cd travel_tool_back` to switch to the project directory
- Run `make start` to install dependencies and start the app
- Login to the travela app. After you are logged in, visit the developer console in your browser and open the `Application` tab, you should see your `jwt-token` under cookies.
- Copy the token and add it to the `authorization` header in `postman`
- In `postman`, make a `POST` request to the endpoint `http://localhost:5000/api/v1/requests/:requestId` with all of following fields as shown in the screenshot. You should get a 201 response with the created request.
- In `postman`, make a `PUT` request to the endpoint `http://localhost:5000/api/v1/requests/:requestId` with all of following fields as shown in the screenshot.
- You should get a 200 response with the updated request

#### Any background context you want to provide?
This tasks works  with notification. To ensure the manager is notified, first create the manager in the Users table.
Also not that a oneWay trip cannot have a return date, a return trip must have a departure date and a return date and  a multi trip must have more than one return trip

#### What are the relevant pivotal tracker stories?
[#160370645](https://www.pivotaltracker.com/story/show/160370645)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-19 at 7 14 48 am" src="https://user-images.githubusercontent.com/25195857/45761816-e6185900-bc24-11e8-8ce5-f4e7c2ab37be.png">
<img width="1440" alt="screen shot 2018-09-19 at 7 16 14 am" src="https://user-images.githubusercontent.com/25195857/45761820-e7498600-bc24-11e8-97af-10fc9b81aab2.png">

#### Questions:
None
